flow/flow.go

```diff
@@ -0,0 +1,303 @@
+package flow
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/micro/go-micro/v2/broker"
+	"github.com/micro/go-micro/v2/client"
+	"github.com/micro/go-micro/v2/logger"
+)
+
+type Flow interface {
```

Comment:
I'm still looking at this interface. Spending time reading about workflow and trying to understand how it should work.

I had previously imagined a "flow" as one thing which would represent a state machine that is executed. In this case it is something of a system that manages many workflows much like a registry, database or broker.

I think you could probably split out Execute/Resume/Pause/Abort/State/Result/Stop into an Executor interface that's then embedded

```
type Flow interface {
    // embedded executor
    Executor
    Init()
    Options()
    String()
}
```

Reply:
You may want to define State as a type which is then returned when checking State so you can provide the sequence of events.

Reply:
Then I think

```
type Flow interface {
   // add a new step
   Add(flow string, step *Step) error
   // update applies against the same step id
   Update(flow string, step *Step) error
   // remove a step
   Remove(flow string, step *Step) error
}
```

Reply:
Alternatively

```
type Flow interface {
   // creates a new flow
   Create(id string) (Task, error)
   // register a task
   Register(id string, Task) error
   // execute a flow
   Execute(id string ...) error
}

Reply:
About Executor i'm fine to exclude it. But for flow interface i need to think more...

Reply:
in case of embeding executor, how to deap that executor can have init and options methods ?

Reply:
> Otherwise Executor could just be passed as an option and Flow only has a Run() method

interesting case